manager/integration/tests/test_basic.py

```diff
@@ -708,6 +709,243 @@ def test_backup_block_deletion(set_random_backupstore, client, core_api, volume_
+    5.  Create backup(1) of the volume.
+    6.  Verify backup block count == 3.
+    7.  Create DR volume from backup(1).
+    8.  Verify DR volume last backup is backup(1).
+    9.  Delete backup(1).
+    10. Verify backup block count == 2.
+    11. Verify DR volume last backup is backup(0).
+    12. Overwrite backup(0) blocks of data on the volume.
+    13. Create backup(2) of the volume.
+    14. Verify DR volume last backup is backup(2).
+    15. Activate and verify DR volume data is data(2).
+    """
+    backupstore_cleanup(client)
+
+    # Set backupstore poll interval to have last backup update quicker
+    update_setting(client, common.SETTING_BACKUPSTORE_POLL_INTERVAL, "5")
+
+    host_id = get_self_host_id()
+
+    vol = create_and_check_volume(client, volume_name, 2, SIZE)
+    vol.attach(hostId=host_id)
+    vol = common.wait_for_volume_healthy(client, volume_name)
+
+    data0 = {'pos': 0, 'len': 2 * BACKUP_BLOCK_SIZE,
+             'content': common.generate_random_data(2 * BACKUP_BLOCK_SIZE)}
+    _, backup0, _, data0 = create_backup(
+        client, volume_name, data0)
+
+    data1 = {'pos': 0, 'len': BACKUP_BLOCK_SIZE,
+             'content': common.generate_random_data(BACKUP_BLOCK_SIZE)}
+    _, backup1, _, data1 = create_backup(
+        client, volume_name, data1)
+
+    backup_blocks_count = backupstore_count_backup_block_files(client,
+                                                               core_api,
+                                                               volume_name)
+    assert backup_blocks_count == 3
+
+    dr_vol_name = "dr-" + volume_name
+    client.create_volume(name=dr_vol_name, size=SIZE,
+                         numberOfReplicas=2, fromBackup=backup1.url,
+                         frontend="", standby=True)
+    check_volume_last_backup(client, dr_vol_name, backup1.name)
+    wait_for_backup_restore_completed(client, dr_vol_name, backup1.name)
+
+    delete_backup(client, volume_name, backup1.name)
+    assert backupstore_count_backup_block_files(client,
+                                                core_api,
+                                                volume_name) == 2
+    check_volume_last_backup(client, dr_vol_name, backup0.name)
+    wait_for_backup_restore_completed(client, dr_vol_name, backup0.name)
+
+    data2 = {'pos': 0,
+             'len': BACKUP_BLOCK_SIZE,
+             'content': common.generate_random_data(BACKUP_BLOCK_SIZE)}
+    _, backup2, _, _ = create_backup(client, volume_name, data2)
+
+    check_volume_last_backup(client, dr_vol_name, backup2.name)
+    wait_for_volume_restoration_start(client, dr_vol_name, backup2.name)
+    wait_for_backup_restore_completed(client, dr_vol_name, backup2.name)
+
+    activate_standby_volume(client, dr_vol_name)
+    dr_vol = client.by_id_volume(dr_vol_name)
+    dr_vol.attach(hostId=host_id)
+    dr_vol = common.wait_for_volume_healthy(client, dr_vol_name)
+    check_volume_data(dr_vol, data2, False)
```

Comment:
The 2nd block of data is not verified.
Same below.

Reply:
The case covers `check_volume_last_backup(client, dr_vol_name, backup1.name)`, I think it should be sufficient.
If I understand correctly, to verify data for block(1) is just to ensure the dr is working? I can add another case just for it so it does not overcomplicate what each case is testing for.

Reply:
After discussing with @shuo-wu , we have agreed there is no need to verify the 2nd block of data because it involves creating 2 dr and we cannot use the validated dr to continue the rest of the case. There is no need to create another case because there is already a test case for that.